### PR TITLE
Adding license header.

### PR DESCRIPTION
**Pull request by elbing**
Fixes mkstemp in stdlib.h

Change-Id: I35b9e46a92190f3bd7b146f8664b51d98af29120

### CODE REVIEW

**Comment by sevki** [comment]
-1

has unrelated stuff.

**Comment by elbing** [comment]
It's trivial, function declaration was missing in stdlib.h, I just reused the branch.

**Comment by elbing** [comment]
Thanks @sevki !

**Comment by sevki** [comment]
LGTM

File: LICENSE.mit [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Copyright (c) 2005-2014 Rich Felker, et al.
+Copyright (c) 2016 Harvey OS
```
Comment (rminnich) [inline LICENSE.mit]: You've got copyright names here and in the files too. This is a bit unusual but it's up to you. Also in this file it's "Harvey OS", whatever that is -- is that a legal entity? If not, why is it here? But even then the name in the files is alvaro. Which is it?

Suggestion: here is what we did in coreboot: 
https://review.coreboot.org/cgit/coreboot.git/tree/

unmodified license. Then, in each file, we put the names with copyrights. That keeps it clear. What you're doing here is mixing up the issue of what the license is vs. who the contributors are and copyright dates.

The name LICENSE for the UCB release license is a questionable choice. It implies it covers the whole tree -- does it? Maybe 
LICENSE.plan9gpl
or something to make it clear it only applies to original Plan 9 sources? You've got lots of stuff in this tree not covered under LICENSE.

File: include/endian.h [hunk 2]
```diff
@@ -1,24 +1,9 @@
...
- *
- * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
- * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
- * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
- * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
- * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
- * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
- * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE.mit file.
```
Comment (rminnich) [inline include/endian.h]: This is good, but general form is to also add a URL for the file as well. I personally don't like putting URLs in files, they tend to go bad, but thought I'd mention it.

**Comment by elbing** [comment]
Ok, License.mit fixed, copyrights in files as they were, old ape license untouched.

**Comment by sevki** [comment]
@rminnich can you LGTM this?

**Comment by rminnich** [comment]
weren't you going to rename to license.musl?

**Comment by sevki** [comment]
@rminnich we good?

**Comment by elbing** [comment]
Ready?

**Pull request merged**